### DISCUSSION TITLE
Environment specific object names

### DISCUSSION THREAD

**Discussion by jackvallis** [Azure-Player/azure.datafactory.tools#269] [Q&A] [answered]
Really love this tool - thanks you very much for your efforts. I have a question on deploying with different Self-Hosted-Integration runtimes. Currently our integration runtimes are set up with environment specific names in ADF, e.g.:
-ADF-DEV-01
-ADF-TEST-01
-ADF-PROD-01

In our previous pipelines (using ARM templates), we had a regex find/replace to replace "ADF-DEV-01" with "ADF-TEST-01"/ "ADF-PROD-01"

I have tried to set the integration runtime names in the config parameters, however this doesn't seem to be the correct approach (error message below). Is there a way to deploy with different integration runtime names across different environments?

Error message trying to set the "name" of the integration runtime:
Wrong path defined in config for object(path):
tegrationRunTime. ADF-DEV-01(properties.Name)

**Comment by NowinskiK** [comment, marked as answer]
Usually, to avoid the problems, the names of the objects should be the same across environments, so no changes.
Therefore, the tool **does not support name changes**. The first node you have access to from csv/json config files is `properties`.
![image](https://user-images.githubusercontent.com/25029325/209137907-ba480c32-945e-4443-88e6-091179253ccd.png)
`name` element can not be addressed.
